datafusion/functions/src/string/common.rs

```diff
@@ -21,17 +21,39 @@ use std::fmt::{Display, Formatter};
 use std::sync::Arc;
 
 use arrow::array::{
-    new_null_array, Array, ArrayAccessor, ArrayDataBuilder, ArrayIter, ArrayRef,
-    GenericStringArray, GenericStringBuilder, LargeStringArray, OffsetSizeTrait,
-    StringArray, StringBuilder, StringViewArray, StringViewBuilder,
+    make_view, new_null_array, Array, ArrayAccessor, ArrayDataBuilder, ArrayIter,
+    ArrayRef, ByteView, GenericStringArray, GenericStringBuilder, LargeStringArray,
+    OffsetSizeTrait, StringArray, StringBuilder, StringViewArray, StringViewBuilder,
 };
 use arrow::buffer::{Buffer, MutableBuffer, NullBuffer};
 use arrow::datatypes::DataType;
+use arrow_buffer::{NullBufferBuilder, ScalarBuffer};
 use datafusion_common::cast::{as_generic_string_array, as_string_view_array};
 use datafusion_common::Result;
 use datafusion_common::{exec_err, ScalarValue};
 use datafusion_expr::ColumnarValue;
 
+/// Make a `u128` based on the given substr, start(offset to view.offset), and
+/// push into to the given buffers
+pub(crate) fn make_and_append_view(
```

Comment:
🤔  I wonder if we should (as a follow on PR) propose adding this upstream to arrow-rs as it seems valuable for any trim related kernels on stringview

Reply:
It sounds great, and https://github.com/apache/datafusion/pull/12383#discussion_r1750580081 can be solved if it is function in arrow-rs.